File: ruby/geometric-classes/lib/square.rb
class Square
  attr_reader :side, :x, :y
  def initialize(side, x=0, y=0)
    @side = side
    @x = x
    @y = y
  end
  def length
    @side
  end
  def width
    @side
  end
  def diameter
    @side * Math::sqrt(2)
  end
  def perimeter
    @side * 4
  end
  def area
    @side ** 2
  end
  def contains_point?(x, y)
    if (x >= @x - (@side*Math.sqrt(2)/2)) && (x <= @x + (@side*Math.sqrt(2)/2)) && (y >= @y - (@side*Math.sqrt(2)/2)) && (y <= @x + (@side*Math.sqrt(2)/2))
      true
    else
      false
    end
  end
end

File: javascript/unit-testing-javascript-with-jasmine/src/Square.js
class Square {
  constructor(length, x, y) {
    this.length = length;
    this.width = length;
    this.x = x || 0;
    this.y = y || 0;
  }

  diameter() {
    var length_squared = Math.pow(this.length, 2);
    var width_squared = Math.pow(this.width, 2);
    return Math.sqrt(length_squared + width_squared);
  };

  perimeter() {
    return this.length * 4;
  };

  area() {
    return Math.pow(this.length, 2);
  };

  containsPoint(x, y) {
    var delta = this.width / 2;
    if(((this.x - delta <= x) && (x <= this.x + delta)) &&
      ((this.y - delta <= y) && (y <= this.y + delta))) {
      return true;
    }
    return false;
  };
}
